app/models/response.rb

```diff
@@ -4,4 +4,54 @@ class Response < ApplicationRecord
   has_many :response_uploaded_files, dependent: :destroy
   has_many :uploaded_files, through: :response_uploaded_files
 
+  def additional_information_key=(value)
+    existing = ai_file
+    if existing
+      update_or_remove_ai_file(value, existing)
+    else
+      create_ai_file(value)
+    end
+  end
+
+  private
+
+  def update_or_remove_ai_file(value, existing)
+    if value.nil?
+      remove_ai_file(existing)
+    else
+      update_ai_file(value, into: existing)
+    end
+  end
+
+  def ai_file
+    uploaded_files.detect { |file| file.filename == 'additional_information.rtf' }
+  end
+
+  def remove_ai_file(existing)
+    uploaded_files.delete(existing)
+  end
+
+  def create_ai_file(value)
```

Comment:
Shouldn't this be in a separate file?

Reply:
I did think about having it in a concern but changed my mind because I wanted to see it as part of the model.  Why do you think it should be in a separate file ?  I was a bit undecided !!

Reply:
I have now moved this into a nice re usable service that can import a file from another bucket into any collection of files